hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconStorageContainerManagerFacade.java

```diff
@@ -396,22 +445,90 @@ private void initializeSCMDB() {
-      LOG.info("Got new checkpoint from SCM : " +
-          dbSnapshot.getCheckpointLocation());
+    if (isSyncDataFromSCMRunning.compareAndSet(false, true)) {
+      DBCheckpoint dbSnapshot = scmServiceProvider.getSCMDBSnapshot();
+      if (dbSnapshot != null && dbSnapshot.getCheckpointLocation() != null) {
+        LOG.info("Got new checkpoint from SCM : " +
+            dbSnapshot.getCheckpointLocation());
+        try {
+          initializeNewRdbStore(dbSnapshot.getCheckpointLocation().toFile());
+        } catch (IOException e) {
+          LOG.error("Unable to refresh Recon SCM DB Snapshot. ", e);
+        }
+      } else {
+        LOG.error("Null snapshot location got from SCM.");
+      }
+    } else {
+      LOG.warn("SCM DB sync is already running.");
+    }
+  }
+
+  public boolean syncWithSCMContainerInfo()
+      throws IOException {
+    if (isSyncDataFromSCMRunning.compareAndSet(false, true)) {
       try {
-        initializeNewRdbStore(dbSnapshot.getCheckpointLocation().toFile());
+        List<ContainerInfo> containers = containerManager.getContainers();
+
+        long containerCount = scmServiceProvider.getContainerCount(
+            HddsProtos.LifeCycleState.CLOSED);
+        // Assumption of size of 1 container info object here is 1 MB
+        long containersMetaDataTotalRpcRespSizeMB =
+            CONTAINER_METADATA_SIZE * containerCount;
+        double numOfApiCalls =
+            Double.valueOf(containersMetaDataTotalRpcRespSizeMB) /
+                Double.valueOf(DEFAULT_RPC_SIZE);
+        long totalApiCalls = Math.round(Math.ceil(numOfApiCalls));
+        long containerCountPerCall = containersMetaDataTotalRpcRespSizeMB <=
+            DEFAULT_RPC_SIZE ? containerCount : DEFAULT_RPC_SIZE;
+        if (containerCount > 0) {
+          for (int startContainerId = 1; startContainerId <= totalApiCalls;
+               startContainerId++) {
+            List<ContainerInfo> listOfContainers = scmServiceProvider.
+                getListOfContainers(Long.valueOf(startContainerId).longValue(),
+                    Long.valueOf(containerCountPerCall).intValue(),
+                    HddsProtos.LifeCycleState.CLOSED);
+            if (null != listOfContainers && listOfContainers.size() > 0) {
+              LOG.info("Got list of containers from SCM : " +
+                  listOfContainers.size());
+              listOfContainers.forEach(containerInfo -> {
+                long containerID = containerInfo.getContainerID();
+                boolean isContainerPresentAtRecon =
+                    containers.remove(containerID);
+                if (!isContainerPresentAtRecon) {
+                  try {
+                    ContainerWithPipeline containerWithPipeline =
+                        scmServiceProvider.getContainerWithPipeline(
+                            containerID);
+                    containerWithPipeline.getContainerInfo().setState(
+                        HddsProtos.LifeCycleState.UNKNOWN);
```

Comment:
Unknown state of container can be removed, its health report to differentiate if Unknown or missing if required.

Reply:
Unknown state reporting is not required, [HDDS-4539](https://issues.apache.org/jira/browse/HDDS-4539) with this when fixed, unknown state will never come.

Reply:
@sumitagrawl - pls re-review. Changes pushed as per comments.